gqcp/include/QCMethod/HF/UHF/UHFFockMatrixCalculation.hpp

```diff
@@ -62,9 +62,10 @@ class UHFFockMatrixCalculation:
 
         const auto& P_alpha = environment.density_matrices_alpha.back();  // the most recent alpha density matrix
         const auto& P_beta = environment.density_matrices_beta.back();    // the most recent beta density matrices
+        SpinResolvedOneDM<Scalar> P {P_alpha, P_beta};
 
-        const auto F_alpha = QCModel::UHF<double>::calculateScalarBasisFockMatrix(Spin::alpha, P_alpha, P_beta, environment.sq_hamiltonian);
-        const auto F_beta = QCModel::UHF<double>::calculateScalarBasisFockMatrix(Spin::beta, P_alpha, P_beta, environment.sq_hamiltonian);
+        const auto F_alpha = QCModel::UHF<double>::calculateScalarBasisFockMatrix(Spin::alpha, P, environment.sq_hamiltonian);
+        const auto F_beta = QCModel::UHF<double>::calculateScalarBasisFockMatrix(Spin::beta, P, environment.sq_hamiltonian);
```

Comment:
The method `QCModel::UHF::calculateScalarBasisFockMatrix` could also return an `USQOneElectronOperator`.